src/audio/module_adapter/module/volume/volume.c

```diff
@@ -572,6 +571,16 @@ static int volume_init(struct processing_module *mod)
 		comp_err(dev, "volume_init(): Failed to allocate %d", vol_size);
 		return -ENOMEM;
 	}
+	/*
+	 * malloc memory to store temp peak volume 4 times to ensure the address
+	 * is 8-byte aligned for multi-way xtensa intrinsic operations.
+	 */
+	cd->peak_vol = rmalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, vol_size);
+	if (!cd->peak_vol) {
+		rfree(cd);
```

Comment:
don't you need to free `cd->vol` here too? Looks like this function's error handling has to be fixed - the error path at new line number 607 below isn't freeing any allocations here. And in general maybe that whole `instance_id` check can be moved to the very top of the function before all the allocations?

Reply:
thanks, done.